test/parallel/test-fs-open.js

```diff
@@ -32,8 +32,15 @@ fs.open(__filename, 'rs', function(err, fd) {
   openFd2 = fd;
 });
 
+assert.throws(function() {
+  fs.open('/path/to/file/that/does/not/exist', { bad: 'flags' }, assert.fail);
+}, function(err) {
+  if (err instanceof TypeError) {
```

Comment:
You want me to assert.equal on err.constructor.name? I'm not sure how else to assert that it's a type error in a one-liner, which seems to be what you're looking for.